predicate/cli/__main__.py

```diff
@@ -14,60 +18,93 @@ def main():
-    obj = policy.export()
-    serialized = yaml.dump(obj)
-    click.echo("deploying policy...")
-    args = ["tctl", "create", "-f"]
-    if sudo:
-        args.insert(0, "sudo")
-
-    subprocess.run(args, text=True, input=serialized, check=True)
-    click.echo(f'policy deployed as resource "{policy.name}"')
+    with open(policy_file, 'r', encoding="utf-8") as file:
+        class_name = get_classname(file.read())
+        module = run_path(policy_file)
+        policy = module[class_name].p
+        click.echo("translating policy...")
+        obj = policy.export()
+        serialized = yaml.dump(obj)
+        click.echo("deploying policy...")
+        args = ["tctl", "create", "-f"]
+        if sudo:
+            args.insert(0, "sudo")
+
+        subprocess.run(args, text=True, input=serialized, check=True)
+        click.echo(f'policy deployed as resource "{policy.name}"')
 
 
 @main.command()
 @click.argument("policy-file")
 def test(policy_file):
-    module = run_path(policy_file)
+    """
+    Test policy
+    """
 
     # Extract the defined policy class and filter out all test functions
-    policyClass = module["Teleport"]
-    fns = {
-        x: y
-        for x, y in policyClass.__dict__.items()
-        if isinstance(y, FunctionType) and x.startswith("test_")
-    }
-
-    # Run all the tests, catching any exceptions and reporting success/failure accordingly
-    click.echo(f"Running {len(fns)} tests:")
-    for name, fn in fns.items():
-        try:
-            fn(policyClass)
-        except Exception as err:
-            out = f"error -> {err}"
-        else:
-            out = "ok"
-
-        click.echo(f"  - {name}: {out}")
+    with open(policy_file, 'r', encoding="utf-8") as file:
+        class_name = get_classname(file.read())
+        module = run_path(policy_file)
+        policy_class = module[class_name]
+        fns = {
+            x: y
+            for x, y in policy_class.__dict__.items()
+            if isinstance(y, FunctionType) and x.startswith("test_")
+        }
+
+        # Run all the tests, catching any exceptions and reporting success/failure accordingly
+        click.echo(f"Running {len(fns)} tests:")
+        for name, functions in fns.items():
+            try:
+                functions(policy_class)
+            except Exception as err:
+                out = f"error -> {err}"
+            else:
+                out = "ok"
+
+            click.echo(f"  - {name}: {out}")
+
+
+@main.command()
+@click.option('--policy', '-p', is_flag=True)
+def new(policy):
+    """
+    Create a new policy based on template
+    """
+    if policy:
+        value = click.prompt('Please enter a policy name', type=str)
+        click.echo("creating policy...")
+        policy = create_policy_from_template(value)
+        # keeping "policies" as a default directory
+        create_dir_if_not_exist("policies")
+        file = open(f"policies/{value.lower()}.py", 'w', encoding="utf-8")
```

Comment:
should we use the with open syntax here for safety and consistency?

Reply:
Agree. Resolved in the following commit.